wherehows-web/app/components/track-ui-event.ts

```diff
@@ -0,0 +1,67 @@
+import Component from '@ember/component';
+import ComputedProperty from '@ember/object/computed';
+import { inject } from '@ember/service';
+import { get, getProperties } from '@ember/object';
+import { action } from 'ember-decorators/object';
+import Metrics from 'ember-metrics';
+import { TrackableEventCategory } from 'wherehows-web/constants/analytics/event-tracking';
+import { IPiwikEvent, TrackableEvent } from 'wherehows-web/typings/app/analytics/event-tracking';
+
+export default class TrackUiEvent extends Component.extend({
```

Comment:
Confused as to why this needs to be a component and not just some injected service